sormas-stats-sitrep/utils/caseExport.R

```diff
@@ -0,0 +1,190 @@
+caseExport <- function(sormas_db, fromDate, toDate){
+  # This function creates two tables as output.
+  # The first table is a line listing of all cases reported in the defined
+  # time period. This line listing of cases includes all the variables as columns
+  # that are necessary for the creation of the SitRep, apart from population and 
+  # geo shapes data.
+  # The second table contains the population and geo data.
+  # It contains data on the population by agegroup and gender in each district.
+  # The two tables are output separately, because each case corresponds to one 
+  # district, but each district corredponds to multiple population counts
+  # (by agegroup and gender). Therefore a line listing by case would be
+  # impractical and also not very suitable for the computation of the 
+  # subsequent parts of the situation report.
+  
+  ## Computing default time based on 365 days in the past if not provided by user
+  if(missing(fromDate) | missing(toDate)){
+    fromDate = as.character(Sys.Date() - delay) 
+    toDate = as.character(Sys.Date()) 
+  }
+
+  ## Loading tables from SORMAS DB
+  
+  # Case table SQL query
+  # (added vaccination status, because it could be interesting for a 
+  # possible extension of the SitRep)
+  queryCases <- paste0("SELECT  DISTINCT id AS case_id,
+    district_id,
+    region_id,
+    caseclassification AS case_classification,
+    EndOfIsolationReason AS case_end_of_iso_reason,
+    reportdate AS report_date_case,
+    disease,
+    vaccinationstatus AS vaccination_status, 
+    symptoms_id,
+    person_id,
+    hospitalization_id 
+  
+    FROM public.cases 
+    
+    WHERE deleted = FALSE 
+    and caseclassification != 'NO_CASE'
+    and reportdate between '", fromDate, "' and '", toDate, "' 
+    and disease = 'CORONAVIRUS' 
+                       ")
+  cases <- dbGetQuery(sormas_db,queryCases)
+  
+  
+  # Person table SQL query
+  queryPersons <- paste0("SELECT DISTINCT id AS person_id,
+    approximateAge AS age_person,
+    sex AS sex_person,
+    deathDate AS death_date_person,
+    causeOfDeathDisease AS cause_of_death_disease_person
+
+    FROM public.person
+
+    WHERE id IN (SELECT person_id FROM public.cases
```

Comment:
This is correct but may not be the most efficient way. You have already selected the case table in the step above , so you can directly use the "cases" table in line 45. This is a sample code:
  queryPerson <- sprintf("SELECT id AS person_id, sex, birthdate_dd, birthdate_mm, birthdate_yyyy
                        FROM public.person
                        WHERE id  in (%s)", paste("'", base::unique(c(case$person_id_case, contact$person_id_contact )), "'",collapse=",") )  # contact$person_id_contact  if contact table is also needed, else delete it.
  person = dbGetQuery(sormas_db, queryPerson)